gapis/api/vulkan/framegraph.go

```diff
@@ -166,6 +212,45 @@ type framegraphInfoHelpers struct {
 	rpInfos  []*renderpassInfo
 	rpInfo   *renderpassInfo
 	currRpId uint64
+	// imageLookup is a lookup table to quickly find images that match a memory
+	// access. Scanning the whole state to look for an image matching a memory
+	// access is slow, this lookup table saves seconds of computation. It is
+	// updated when a new renderpass is started under a new top-level parent
+	// command: images can only be created/destroyed between top-level commands,
+	// not during the execution of subcommands.
+	imageLookup  map[memory.PoolID]map[memory.Range][]*ImageObjectʳ
+	parentCmdIdx uint64 // used to know when to update imageLookup
+}
+
+// updateImageLookup updates the lookup table to quickly find an image matching a memory observation.
+func (helpers *framegraphInfoHelpers) updateImageLookup(state *State) {
+	helpers.imageLookup = make(map[memory.PoolID]map[memory.Range][]*ImageObjectʳ)
```

Comment:
This looks like a very deeply accessed information that can cause consecutive cache misses(e.g accessing the elements of two layered map). If there is no alternative, then it's fine but otherwise, for the future, this structure can be reconsidered.

Reply:
This of course depends how it's being read.

Reply:
In practice, given how memory accesses are recorded at capture time, there is almost always only one memory range associated to a memory pool. In the lookup, we know the poolID so we directly access to the second-level map, then we iterate over it and it's not a big deal since it will almost always have only one element.